src/components/SettingsModal.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import SettingsPage from '../pages/SettingsPage';
+import CONST from '../CONST';
+import themeColors from '../styles/themes/default';
+import ONYXKEYS from '../ONYXKEYS';
+import ModalWithHeader from './ModalWithHeader';
+import {redirect} from '../libs/actions/App';
+import ROUTES from '../ROUTES';
+
+/**
+ * Right-docked modal view showing a user's settings.
+ */
+const propTypes = {
+    // Is the Settings Modal visible or not?
+    isVisible: PropTypes.bool,
+
+    /* Onyx Props */
+    // Currently viewed reportID
+    currentlyViewedReportID: PropTypes.string,
+};
+
+const defaultProps = {
+    isVisible: false,
+    currentlyViewedReportID: '',
+};
+
+const SettingsModal = props => (
+    <ModalWithHeader
+        type={CONST.MODAL.MODAL_TYPE.RIGHT_DOCKED}
+        onClose={() => redirect(ROUTES.getReportRoute(props.currentlyViewedReportID))}
```

Comment:
This seems like a minor regression from what we were doing before. Which is, cover the case where `currentlyViewedReportID` has no value.

Reply:
Ah yep, missed that - nice catch.